src/v/kafka/types.h

```diff
@@ -91,6 +92,33 @@ using leader_epoch = named_type<int32_t, struct leader_epoch_tag>;
  */
 static constexpr leader_epoch invalid_leader_epoch(-1);
 
+struct uuid {
+    static constexpr auto length = 16;
+
+    template<typename Consumer>
+    void set_uuid(Consumer& c) {
+        c.consume(length, [this](const char* src, size_t max) {
+            std::copy_n(src, max, _uuid.data());
+            return ss::stop_iteration::no;
+        });
+    }
+
+    /// implicit conversions to string_view
+    std::string_view view() const {
+        return std::string_view(
+          reinterpret_cast<const char*>(_uuid.data()), _uuid.size());
+    }
+
+    ss::sstring to_string() const { return base64_to_string(view()); }
```

Comment:
Is a `uuid` base64 encoded?
```suggestion
    ss::sstring to_string() const { return to_hex(view()); }
```
But typically `operator<<` or `fmt::formatter` are nicer, since it'll be greater than 15 chars and need an allocation.

Reply:
+1 to the suggestion, & no they are not base64 encoded

Reply:
Ok, in that case:
```suggestion
    ss:sstring to_string() const { return bytes_to_base64(view()); }
```